VoteStakeRegistry/sdk/client.ts

```diff
@@ -1,20 +1,108 @@
-import { Program, Provider, web3 } from '@coral-xyz/anchor'
+import { Program, Provider, web3} from '@coral-xyz/anchor'
 import { IDL, VoterStakeRegistry } from './voter_stake_registry'
+import {Client} from "@solana/governance-program-library";
+import {PublicKey, TransactionInstruction} from "@solana/web3.js";
+import {getVoterPDA} from "./accounts";
+import {SYSTEM_PROGRAM_ID} from "@solana/spl-governance";
+import BN from "bn.js";
+import {fetchVotingPower} from "@hooks/queries/plugins/vsr";
 
 export const DEFAULT_VSR_ID = new web3.PublicKey(
   'vsr2nfGVNHmSY8uxoBGqq8AQbwz3JwaEaHqGbsTPXqQ'
 )
 
-export class VsrClient {
-  constructor(
-    public program: Program<VoterStakeRegistry>,
-    public devnet?: boolean
-  ) {}
+export class VsrClient extends Client<typeof IDL> {
+  readonly requiresInputVoterWeight = false;
+
+  getRegistrarPDA(realm: PublicKey, mint: PublicKey) {
+    const [registrar, registrarBump] = PublicKey.findProgramAddressSync(
+        [realm.toBuffer(), Buffer.from('registrar'), mint.toBuffer()],
+        this.program.programId
+    )
+    return {
+      registrar,
+      registrarBump,
+    }
+  }
+
+  getVoterWeightRecordPDA(realm: PublicKey, mint: PublicKey, walletPk: PublicKey) {
+    const {registrar} = this.getRegistrarPDA(realm, mint);
+
+    const [voterWeightPk, voterWeightRecordBump] = PublicKey.findProgramAddressSync(
+        [
+          registrar.toBuffer(),
+          Buffer.from('voter-weight-record'),
+          walletPk.toBuffer(),
+        ],
+        this.program.programId
+    )
+
+    return {
+      voterWeightPk,
+      voterWeightRecordBump,
+    }
+  }
+
+  async getVoterWeightRecord() {
+    return null;
+  }
+
+  // NO-OP TODO: How are Vsr voter weight records created?
+  async createVoterWeightRecord(): Promise<TransactionInstruction | null> {
```

Comment:
is this ok?

Reply:
Apologies, I didn't scour the PR for TODOs properly it seems. This is also answered now, and fixed in my latest commit.